### PR TITLE
Add tests for help, tracksubject and sort functions

### LINKED ISSUE #95: Add tests for SortCommand

### LINKED ISSUE #108: Add tests for Help function

### LINKED ISSUE #155: Add tests for tracksubject

### PR DESCRIPTION
**Pull request by adriantan101**
Fixes #95 
Fixes #108 
Fixes #155

### CODE REVIEW

File: src/main/java/seedu/academyassist/model/person/SubjectEnum.java [hunk 1]
```diff
@@ -39,10 +39,15 @@ public String getSubjectName() {
      * Checks if the given subject is valid.
      * A valid subject matches any of the available subjects, case-insensitive.
      *
-     * @param subject The subject to check.
+     * @param subject The subject to check. Must not be null.
      * @return {@code true} if the subject is valid, {@code false} otherwise.
+     * @throws NullPointerException if the subject parameter is null
      */
     public static boolean isValidSubject(String subject) {
+        if (subject == null) {
```
Comment (annabellekk) [inline src/main/java/seedu/academyassist/model/person/SubjectEnum.java:47]: Good that you handle null instance of subject

File: src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        model = new ModelManager(getTypicalAcademyAssist(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_sortByName_correctOrdering() throws CommandException {
+        SortCommand sortCommand = new SortCommand(new SortParam("name"));
+        sortCommand.execute(model);
+
+        List<Person> sortedList = model.getFilteredPersonList();
+        // Check if list has at least 2 elements to compare
```
Comment (annabellekk) [inline src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java:37]: Nice that you added comments to aid understanding

File: src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+                String nextId = sortedList.get(i + 1).getStudentId().toString();
+                assertTrue(currentId.compareTo(nextId) <= 0,
+                        String.format("Student ID at index %d (%s) should come before or equal to ID at index %d (%s)",
+                                i, currentId, i + 1, nextId));
+            }
+        }
+    }
+
+    @Test
+    public void execute_emptyList_success() throws CommandException {
```
Comment (annabellekk) [inline src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java:84]: Good that you handle empty list as well

File: src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        Model emptyModel = new ModelManager();
+        SortCommand sortCommand = new SortCommand(new SortParam("name"));
+        CommandResult result = sortCommand.execute(emptyModel);
+        assertEquals(String.format(SortCommand.MESSAGE_SUCCESS, "name"), result.getFeedbackToUser());
+    }
+
+    @Test
+    public void execute_singleItemList_remainsUnchanged() throws CommandException {
```
Comment (annabellekk) [inline src/test/java/seedu/academyassist/logic/commands/SortCommandIntegrationTest.java:92]: Good catch of edge cases

File: src/test/java/seedu/academyassist/logic/commands/SortCommandTest.java [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    public void execute_sortById_success() {
+        SortCommand sortCommand = new SortCommand(new SortParam("studentId"));
+        String expectedMessage = String.format(SortCommand.MESSAGE_SUCCESS, "studentId");
+        expectedModel.sortAcademyAssistById();
+
+        assertCommandSuccess(sortCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void constructor_nullSortParam_throwsNullPointerException() {
```
Comment (annabellekk) [inline src/test/java/seedu/academyassist/logic/commands/SortCommandTest.java:58]: Good catch of potential misuse with null case

File: src/test/java/seedu/academyassist/model/person/SubjectTest.java [hunk 6]
```diff
@@ -18,8 +19,46 @@ public void constructor_invalidSubjectName_throwsIllegalArgumentException() {
...
+
+        // null -> returns false
+        assertFalse(english.equals(null));
+
+        // different type -> returns false
+        assertFalse(english.equals(5));
+
+        // different subject -> returns false
+        Subject math = new Subject("Math");
+        assertFalse(english.equals(math));
```
Comment (JoanneHing) [inline src/test/java/seedu/academyassist/model/person/SubjectTest.java:62]: Thanks for including the comment to show what is the test cases testing!

File: src/test/java/seedu/academyassist/logic/parser/SortCommandParserTest.java [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        assertParseFailure(parser, "name",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, SortCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_extraArguments_throwsParseException() {
+        assertParseFailure(parser, " " + PREFIX_SORT_PARAM + "name extra",
+                MESSAGE_CONSTRAINTS);
+    }
+}
```
Comment (JoanneHing) [inline src/test/java/seedu/academyassist/logic/parser/SortCommandParserTest.java:53]: Maybe can include some comment here to show what is the test cases testing here too!

File: src/test/java/seedu/academyassist/logic/commands/CommandResultTest.java [hunk 8]
```diff
@@ -3,20 +3,35 @@
...
     @Test
     public void equals() {
         CommandResult commandResult = new CommandResult("feedback");
 
         // same values -> returns true
         assertTrue(commandResult.equals(new CommandResult("feedback")));
-
         assertTrue(commandResult.equals(new CommandResult("feedback", false, false, false,
-                                                          false, null)));
+                false, null)));
```
Comment (JoanneHing) [inline src/test/java/seedu/academyassist/logic/commands/CommandResultTest.java:34]: Good that you notice the styling error.

**Review by JoanneHing — APPROVED** [review]
